src/_P042_Candle.ino

```diff
@@ -76,422 +83,440 @@ enum ColorType {
+        Candle_bright = BRIGHT_START;
+      }
+      Candle_type  = static_cast<SimType>(PCONFIG(4));
+      Candle_color = static_cast<ColorType>(PCONFIG(5));
+
+      if (!Candle_pixels || (GPIO_Set == false))
       {
-        Candle_red = PCONFIG(0);
-        Candle_green = PCONFIG(1);
-        Candle_blue = PCONFIG(2);
-        Candle_bright = PCONFIG(3);
-        if (Candle_red == 0 && Candle_green == 0 && Candle_blue == 0) {
-          Candle_bright = BRIGHT_START;
-        }
-        Candle_type = (SimType)PCONFIG(4);
-        Candle_color = (ColorType)PCONFIG(5);
+        GPIO_Set = validGpio(CONFIG_PIN1);
 
-        if (!Candle_pixels || GPIO_Set == false)
-        {
-          GPIO_Set = validGpio(CONFIG_PIN1);
-          if (Candle_pixels) {
-            delete Candle_pixels;
-          }
-          Candle_pixels = new (std::nothrow) Adafruit_NeoPixel(NUM_PIXEL, CONFIG_PIN1, NEO_GRB + NEO_KHZ800);
-          if (Candle_pixels != nullptr) {
-            SetPixelsBlack();
-            Candle_pixels->setBrightness(Candle_bright);
-            Candle_pixels->begin();
-          }
+        if (Candle_pixels) {
+          delete Candle_pixels;
+        }
+        Candle_pixels = new (std::nothrow) Adafruit_NeoPixel(NUM_PIXEL, CONFIG_PIN1, NEO_GRB + NEO_KHZ800);
 
-          #ifndef BUILD_NO_DEBUG
-          if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
-            String log = F("CAND : Init WS2812 Pin : ");
-            log += CONFIG_PIN1;
-            addLogMove(LOG_LEVEL_DEBUG, log);
-          }
-          #endif
+        if (Candle_pixels != nullptr) {
+          SetPixelsBlack();
+          Candle_pixels->setBrightness(Candle_bright);
+          Candle_pixels->begin();
         }
 
-        success = Candle_pixels != nullptr;
-        break;
+          # ifndef BUILD_NO_DEBUG
+
+        if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
+          addLogMove(LOG_LEVEL_DEBUG, concat(F("CAND : Init WS2812 Pin : "), static_cast<int>(CONFIG_PIN1)));
+        }
+          # endif // ifndef BUILD_NO_DEBUG
       }
 
+      success = Candle_pixels != nullptr;
+      break;
+    }
+
     case PLUGIN_EXIT:
-      {
-        if (Candle_pixels != nullptr) {
-          delete Candle_pixels;
-          Candle_pixels = nullptr;
-        }
-        break;
+    {
+      if (Candle_pixels != nullptr) {
+        delete Candle_pixels;
+        Candle_pixels = nullptr;
       }
+      break;
+    }
 
     case PLUGIN_ONCE_A_SECOND:
-      {
-        Candle_pixels->setBrightness(Candle_bright);
-        Candle_pixels->show(); // This sends the updated pixel color to the hardware.
-        success = true;
-        break;
-      }
+    {
+      Candle_pixels->setBrightness(Candle_bright);
+      Candle_pixels->show(); // This sends the updated pixel color to the hardware.
+      success = true;
+      break;
+    }
 
     case PLUGIN_FIFTY_PER_SECOND:
+    {
+      switch (Candle_type)
       {
-        switch (Candle_type)
+        case 0: // "Update" for OFF
```

Comment:
Why not making #defines for these enum values?
Then the comments can also be removed.

Reply:
Fixed with `enum class` definition